packages/cli/lib/validators/marketplaceValidators/theme/DependencyValidator.js

```diff
@@ -0,0 +1,119 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+const { HUBL_EXTENSIONS } = require('@hubspot/cli-lib/lib/constants');
+const { validateHubl } = require('@hubspot/cli-lib/api/validate');
+const {
+  getDepsFromHublValidationObject,
+} = require('@hubspot/cli-lib/validate');
+const { getExt, isRelativePath } = require('@hubspot/cli-lib/path');
+
+const BaseValidator = require('../BaseValidator');
+
+const MISSING_ASSET_CATEGORY_TYPES = ['MODULE_NOT_FOUND'];
+
+class DependencyValidator extends BaseValidator {
+  constructor(options) {
+    super(options);
+
+    this.errors = {
+      EXTERNAL_DEPENDENCY: {
+        key: 'externalDependency',
+        getCopy: ({ file, path }) =>
+          `${file} contains a path that points to an external dependency (${path})`,
+      },
+      ABSOLUTE_DEPENDENCY_PATH: {
+        key: 'absoluteDependencyPath',
+        getCopy: ({ file, path }) =>
+          `${file} contains an absolute path (${path})`,
+      },
+    };
+  }
+
+  // HACK We parse paths from rendering errors because the renderer won't
+  // include paths in the all_dependencies object if it can't locate the asset
+  getPathsFromRenderingErrors(validation) {
+    return validation.renderingErrors
+      .filter(renderingError =>
+        MISSING_ASSET_CATEGORY_TYPES.includes(renderingError.category)
+      )
+      .map(renderingError => renderingError.categoryErrors.path);
+  }
+
+  async getAllDependenciesByFile(files, accountId) {
+    return Promise.all(
+      files
+        .filter(file => HUBL_EXTENSIONS.has(getExt(file)))
+        .map(async file => {
+          const source = await fs.readFile(file, { encoding: 'utf8' });
+          if (!(source && source.trim())) {
+            return { file, deps: {} };
+          }
+          const validation = await validateHubl(accountId, source);
+          const deps = getDepsFromHublValidationObject(validation);
+          deps.RENDERING_ERROR_PATHS = this.getPathsFromRenderingErrors(
+            validation
+          );
+          return { file, deps };
+        })
+    );
+  }
+
+  isExternalDep(absoluteThemePath, file, relativeDepPath) {
```

Comment:
I commented this to make it less confusing. Let me know if there's an easier way to accomplish this `isExternalDep` check!